prow/plugins/trigger/push.go

```diff
@@ -18,15 +18,25 @@ package trigger
 
 import (
 	"context"
+	"k8s.io/test-infra/prow/git/v2"
 
 	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
 	"k8s.io/test-infra/prow/config"
 	"k8s.io/test-infra/prow/github"
 	"k8s.io/test-infra/prow/pjutil"
 )
 
-func listPushEventChanges(pe github.PushEvent) config.ChangedFilesProvider {
+func listPushEventChanges(gc git.ClientFactory, pe github.PushEvent) config.ChangedFilesProvider {
 	return func() ([]string, error) {
+		if gc != nil {
```

Comment:
I would use this as a fallback method

Reply:
The thing is Idk what are the consequences of this decision and what would be the difference, but I know trigger is quite important plugin and I do not want to break it's compatibility, so either you properly test the code and prove your solution is a good a valid replacement or you just execute github version, you fail because of the limit you mentioned and then renew using git client.

Reply:
> but I know trigger is quite important plugin and I do not want to break it's compatibility

:100: 

>  The github version was working up until now for most of cases.

Agree, of course!

>  So either you properly test the code and prove your solution is a good a valid replacement or you just execute github version, you fail because of the limit you mentioned and then renew using git client.

If only there was a way to know whether Github truncated its event, it would be fairly easy. 
I will dig into that, thank you very much!